includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1710,6 +1719,17 @@ public function process_payment_for_order( $cart, $payment_information, $schedul
 
 		$this->maybe_add_customer_notification_note( $order, $processing );
 
+		if ( isset( $status ) && Intent_Status::REQUIRES_ACTION === $status && $this->is_changing_payment_method_for_subscription() ) {
+			// Because we're filtering woocommerce_subscriptions_update_payment_via_pay_shortcode, we need to manually set this delayed update all flag here.
+			if ( isset( $_POST['update_all_subscriptions_payment_method'] ) && wc_clean( wp_unslash( $_POST['update_all_subscriptions_payment_method'] ) ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Missing
+				$order->update_meta_data( '_delayed_update_payment_method_all', wc_clean( wp_unslash( $_POST['payment_method'] ) ) ); // phpcs:ignore WordPress.Security.NonceVerification.Missing
+				$order->save();
+			}
+
+			wp_safe_redirect( $response['redirect'] );
+			exit;
+		}
+
```

Comment:
This is to set the stage for 3DS handling and force quit to stop subscriptions from making unwanted changes, as we will handle once 3DS is confirmed later by `update_order_status` AJAX.